hazelcast/src/main/java/com/hazelcast/query/impl/predicates/LikePredicate.java

```diff
@@ -43,6 +48,28 @@ public LikePredicate(String attributeName, String expression) {
         this.expression = expression;
     }
 
+    @Override
+    public Set<QueryableEntry> filter(QueryContext queryContext) {
+        Index index = queryContext.getIndex(attributeName);
+        if (indexPrefix == null) {
```

Comment:
Caching of the index prefix doesn't make much sense here: `filter` is usually called only once during the predicate lifetime.

Reply:
addressed in https://github.com/hazelcast/hazelcast/pull/18289/commits/7e2695ce41ad1208994dafd70be01b14118a792e